src/diff/oasDiff.ts

```diff
@@ -14,71 +189,327 @@ import { execSync } from "child_process";
+  let hasErrors = false;
+
+  // Find all exchange.json files and their parent directories
+  const baseExchangeDirs = await findExchangeDirectories(baseApi);
+  const newExchangeDirs = await findExchangeDirectories(newApi);
+
+  const allDirNames = new Set([
+    ...baseExchangeDirs.map((dir) => dir.name),
+    ...newExchangeDirs.map((dir) => dir.name),
+  ]);
+
+  for (const dirName of allDirNames) {
+    const baseDir = baseExchangeDirs.find((dir) => dir.name === dirName);
+    const newDir = newExchangeDirs.find((dir) => dir.name === dirName);
+
+    // Check if directory was deleted
+    if (baseDir && !newDir) {
+      console.log(`${dirName} API is deleted`);
+      if (flags.format === "json") {
+        allResults.push({
+          directory: dirName,
+          status: "deleted",
+          message: `${dirName} API is deleted`,
+        });
+      } else {
+        allResults.push(`======${dirName} API is deleted======`);
+      }
+      hasChanges = true;
+      continue;
+    }
+
+    // Check if directory was added
+    if (!baseDir && newDir) {
+      console.log(`${dirName} API is added`);
+      if (flags.format === "json") {
+        allResults.push({
+          directory: dirName,
+          status: "added",
+          message: `${dirName} API is added`,
+        });
+      } else {
+        allResults.push(`======${dirName} API is added======`);
+      }
+      hasChanges = true;
+      continue;
+    }
+
+    // Both directories exist, compare individual yml files in each directory
+    if (baseDir && newDir) {
+      console.log("===================================");
+      console.log(`Processing directory pair: ${dirName}`);
+
+      try {
+        const baseYamlFiles = await findYamlFiles(baseDir.path);
+        const newYamlFiles = await findYamlFiles(newDir.path);
+
+        const directoryChanges: (FileChange | Record<string, unknown>)[] = [];
+        const directoryChangesText: string[] = [];
+
+        // Process each YAML file pair
+        for (const baseYamlFile of baseYamlFiles) {
+          const baseFileName = path.basename(baseYamlFile);
+          const newYamlFile = newYamlFiles.find(
+            (f) => path.basename(f) === baseFileName
+          );
+
+          if (newYamlFile) {
+            console.log(`Comparing ${baseFileName} in ${dirName}`);
+            const oasdiffOutput = await executeOasDiff(
+              baseYamlFile,
+              newYamlFile,
+              jsonMode
+            );
+
+            if (oasdiffOutput.trim().length > 0) {
+              if (flags.format === "json") {
+                const outputJson = JSON.parse(oasdiffOutput);
+                if (outputJson?.length > 0) {
+                  directoryChanges.push(...outputJson);
+                }
+              } else {
+                directoryChangesText.push(
+                  `--- Changes in ${baseFileName} ---\n${oasdiffOutput}`
+                );
+              }
+            }
+          } else {
```

Comment:
nit: Is this case even possible.. where the directory exists but Yaml is missing? May be when download has some issues

Reply:
Good point. I think it is stil good to leave it here if download happens to has some issue and dev does not realize it and run oasdiff. The logic in here simply notify which yml files are removed from previous version into output file